cql3/expr/expression.cc

```diff
@@ -1082,6 +1082,59 @@ std::ostream& operator<<(std::ostream& s, oper_t op) {
     __builtin_unreachable();
 }
 
+std::vector<expression> extract_single_column_restrictions_for_column(const expression& expr,
+                                                                      const column_definition& column) {
+    struct visitor {
```

Comment:
We should aim to have this and similar functions have two returns, one of the extracted single column restriction, the other for everything else. The "everything else" can be further split if there are restrictions on a clustering key prefix, and the remainder is used for filtering if it's not a constant true.

Reply:
Ideally, such split would happen early (say in statement_restrictions constructor) so it wouldn't be necessary to do here.  More ideally, the bounds/filter calculation would be bytecode generated at query-planning time, serving the surrounding bytecode for accessing tables and index indirection.